### PR TITLE
DSL to perform assertions on eventshub store

### LINKED ISSUE #42: Rework the eventshub.Store assertion apis
The `eventshub.Store` apis at the moment are just a port of `recordevents.EventInfoStore` from eventing. We need to adapt the assertion/matcher interfaces to:

* Adapt to the `StepFn` contract (do we need it?)
* @n3wscott comment https://github.com/knative-sandbox/reconciler-test/pull/41#discussion_r525579156
* @n3wscott comment https://github.com/knative-sandbox/reconciler-test/pull/41#discussion_r525582874

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Fix #42

Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

# Changes

- :gift: DSL to perform assertions on eventshub store. This DSL is a beginning, we might extend in future to have other assertions on the event store

### CODE REVIEW

File: test/example/recorder_feature.go [hunk 1]
```diff
@@ -17,15 +17,15 @@ limitations under the License.
...
-	"context"
-	"testing"
 	"time"
 
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"knative.dev/reconciler-test/pkg/k8s"
 
 	. "github.com/cloudevents/sdk-go/v2/test"
 
+	. "knative.dev/reconciler-test/pkg/eventshub"
```
Comment (n3wscott) [inline test/example/recorder_feature.go:27]: I would rather not dot import this to make it more clear what is happening. This is an example meant to be copied and the dot import is a bit of magic that makes it unclear which import is bringing those methods

File: test/example/recorder_feature.go [hunk 2]
```diff
@@ -47,9 +45,8 @@ func RecorderFeature() *feature.Feature {
 
 	f.Alpha("direct sending between a producer and a recorder").
 		Must("the recorder received all sent events within the time",
-			func(ctx context.Context, t *testing.T) {
-				eventshub.StoreFromContext(ctx, to).AssertExact(1, eventshub.MatchEvent(HasId(event.ID())))
-			})
+			eventshub.OnStore(to).Exact(1).Match(eventshub.MatchEvent(HasId(event.ID()))),
```
Comment (tayarani) [inline test/example/recorder_feature.go:48]: When reading this, shouldn't it be: OnStore().Match(...).Exact(1) ? Or is this an impossible change at this stage?

Comment (slinkydeveloper) [reply test/example/recorder_feature.go:48]: I think this is doable, i have no preference :smile:

Comment (slinkydeveloper) [reply test/example/recorder_feature.go:48]: It's even better because you can chain separate stuff, i'm on it

Comment (slinkydeveloper) [reply test/example/recorder_feature.go]: It also made the code simpler!

**Comment by tayarani** [comment]
This looks a lot cleaner. Awesome!

**Pull request merged**